crates/sema/src/ast_passes.rs

```diff
@@ -25,6 +25,7 @@ struct AstValidator<'sess, 'ast> {
     function_kind: Option<ast::FunctionKind>,
     in_unchecked_block: bool,
     in_loop_depth: u64,
+    x_depth: u64, // How far away are you from the nearest loop
```

Comment:
I think it's simpler to just have a separate function for loop bodies that check if they're a declaration statement and emit the error there

Reply:
✅

Reply:
I have removed `x_depth` because I have moved it to a new function in `utils.rs`